src/app/public/modules/modal/modal.interface.ts

```diff
@@ -1,10 +1,69 @@
+// TODO: defaults won't show in the generated docs until this work is done:
+// https://github.com/blackbaud/skyux-docs-tools/issues/38
+
+/**
+ * Specifies configuration options for creating a modal.
+ */
 export interface SkyModalConfigurationInterface {
+
+  /**
+   * Indicates whether to display the modal full screen.
+   * This property defaults to `false`.
+   */
   fullPage?: boolean;
+
+  /**
+   * Specifies a size for the modal. The valid options are `small`, `medium`, and `large`.
+   * If no value is provided, the size will be set to `medium`.
+   */
   size?: string;
+
+  /**
+   * An array property of `providers`.
+   * In Angular, a provider is something that can create or deliver a service.
+   * This property can be used to pass context values from the component that launches the modal to the modal component.
+   */
   providers?: any[];
+
+  /**
+   * Sets the `aria-describedby` attribute for the modal dialog to support accessibility.
+   * The value should be an ID (without the leading `#`) that points to the element
+   * that describes the modal. Typically, this is the text on the modal,
+   * not including anything that users interact with such as buttons or a form.
+   * If no ID is specified, the default value is the content of the `sky-modal-content` component.
+   */
   ariaDescribedBy?: string;
+
+  /**
+   * Sets the `aria-labelledby` attribute for the modal dialog to support accessibility.
+   * The value should be an ID (without the leading `#`) that points to the element
+   * that labels the modal. Typically this is a header element. If no ID is specified,
+   * the default value is the content of the `sky-modal-header` component.
+   */
   ariaLabelledBy?: string;
+
+  /**
+   * Specifies an ARIA role for the modal dialog to support accessibility
```

Comment:
I would make `an ARIA role` the link to www.w3.org/WAI/PF/aria/roles and delete the "For information..." sentence to tighten this up.